tests/conftest.py

```diff
@@ -1026,19 +1069,315 @@ def get_state_db(self):
+                for key in intfCfg:
+                    nbraddr = ""
+                    if key.lower().startswith(nbrintf):
+                        intfaddr = re.split("/|\\|", key)
+                        if len(intfaddr) > 1:
+                            nbraddr = intfaddr[1]
+                        if nbraddr == "":
+                            continue
+                        if ctnname not in neighs_by_ctn:
+                            neighs_by_ctn[ctnname] = []
+                        neighs_by_ctn[ctnname].append((nbrvethintf - 1, nbraddr))
+        return neighs_by_ctn
+
+    def handle_neighconn(self):
+        if self.oper != "create":
+            return
+        neighs_by_ctn = self.get_topo_neigh()
+        for ctnname, nbraddrs in neighs_by_ctn.iteritems():
+            if ctnname not in self.dvss:
+                continue
+            for idx, nbraddr in nbraddrs:
+                self.dvss[ctnname].servers[idx].runcmd("ifconfig eth0 down ")
+                self.dvss[ctnname].servers[idx].runcmd("ifconfig eth0 up")
+                self.dvss[ctnname].servers[idx].runcmd("ifconfig eth0 " + nbraddr)
+        return
+
+    def verify_chassdb(self):
+        passed = True
+        cwd = os.getcwd()
+        ctn = self.virt_topo["chassis_instances"]
+        for ctndir in ctn:
+            cfgdir = cwd + "/virtual_chassis/" + ctndir
+            cfgfile = cfgdir + "/default_config.json"
+            with open(cfgfile, "r") as cfg:
+                defcfg = json.load(cfg)["DEVICE_METADATA"]["localhost"]
+                ctnname, _ = self.get_ctninfo(ctndir)
+                # verify connectivity to chassis-db from ctn
+                if "chassis_db_address" in defcfg:
+                    print("verify chassisdb connectivity to %s " % ctnname)
+                    _, out = self.runcmd_on_ctn(ctnname,
+                                    "sonic-db-cli CHASSIS_DB keys SYSTEM_PORT\"*\"")
+                    if len(out.split("\n")) != 97:
+                        print("FAILED:chassisdb in vs must have %s sysports" % out)
+                        print(len(out.split("\n")))
+                        passed = False
+        return passed
+
+    def verify_conns(self):
+        passed = True
+        if "neighbor_connections" not in self.virt_topo:
+            return passed
+        neighs_by_ctn = self.get_topo_neigh()
+        for ctnname, nbraddrs in neighs_by_ctn.iteritems():
+            for item in nbraddrs:
+                nbraddr = item[1]
+                print("verify neighbor connectivity from %s to %s nbrAddr " % (
+                   ctnname, nbraddr))
+                _, out = self.runcmd_on_ctn(ctnname, " ping -c 5 " + nbraddr)
+                if "5 received" not in out.split("\n")[-3]:
+                    print("FAILED:%s: ping %s \n res: %s " % (ctnname, nbraddr, out))
+                    passed = False
+        return passed
+
+    def verify_crashes(self):
```

Comment:
[Not a review comment] @lguohan we should def have something like this for pizza-box